source/aws_signing.c

```diff
@@ -1707,6 +1707,90 @@ static int s_build_canonical_request_hash(struct aws_signing_state_aws *state) {
     return result;
 }
 
+static int s_build_canonical_request_event(struct aws_signing_state_aws *state) {
+    int result = AWS_OP_ERR;
+
+    struct aws_byte_buf *dest = &state->string_to_sign_payload;
+
+    /* previous signature + \n */
+    struct aws_byte_cursor prev_signature_cursor;
+    AWS_ZERO_STRUCT(prev_signature_cursor);
+    if (aws_signable_get_property(state->signable, g_aws_previous_signature_property_name, &prev_signature_cursor)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_SIGNING, "(id=%p) Event signable missing previous signature property", (void *)state->signable);
+        return aws_raise_error(AWS_AUTH_SIGNING_MISSING_PREVIOUS_SIGNATURE);
+    }
+
+    /* strip any padding (AWS_SIGV4A_SIGNATURE_PADDING_BYTE) from the previous signature */
+    prev_signature_cursor = aws_trim_padded_sigv4a_signature(prev_signature_cursor);
+
+    if (aws_byte_buf_append_dynamic(dest, &prev_signature_cursor)) {
+        return AWS_OP_ERR;
+    }
+
+    if (aws_byte_buf_append_byte_dynamic(dest, '\n')) {
+        return AWS_OP_ERR;
+    }
+
+    /* Append hex sha encoded date header */
+    const struct aws_signing_config_aws *config = &state->config;
+
```

Comment:
trivial: maybe just remove this variable. It's weird to be like "OK NOW WE'RE DOING THE DATE HEADER" and then make this variable and then again be like "OK NOW WE'RE DOING THE DATE HEADER AND HERE'S HOW WE'RE DOING IT" and you don't use this variable until 20 lines later
```suggestion
```

Reply:
Thanks, fixed.